File: 2-3-3.py
import sys
input=sys.stdin.readline

def solution(A, B):
    D = {}
    for b in B:
        if b in D:
            D[b] += 1
        else:
            D[b] = 1

    answer = []
    for a in A:
        if a not in D:
            answer.append(a)
    answer.sort()
    return answer

A = list(input().split())
B = list(input().split())
C = solution(A, B)
for c in C:
    print(c)

File: 2-3-3.cpp
#include<bits/stdc++.h>
using namespace std;
int main() {
    vector<string> A;
    do {
        string s; cin >> s;
        A.push_back(s);
    } while (getc(stdin) == ' ');

    vector<string> B;
    do {
        string s; cin >> s;
        B.push_back(s);
    } while (getc(stdin) == ' ');

    map<string, int> D;
    for (auto& b : B) {
        D[b]++;
    }

    vector<string> answer;
    for (auto& a : A) {
        if (D[a] == 0) {
            answer.push_back(a);
        }
    }
    sort(answer.begin(), answer.end());
    for (auto& a : answer) {
        cout << a << '\n';
    }
}